### PR TITLE
[SOL-22] and [SOL-23] Implement CLI logging and exit code

### PR DESCRIPTION
**Pull request by tli-ocient**
https://ocient.atlassian.net/browse/SOL-22
https://ocient.atlassian.net/browse/SOL-23

### CODE REVIEW

File: src/main/java/com/ocient/cli/CLI.java [hunk 1]
```diff
@@ -91,6 +93,7 @@ private static void connectTo(final String cmd)
 			{
 				System.out.println("Error: " + e.getMessage());
 				lastCommandErrored = true;
+				returnErrorCode = true;
```
Comment (dcoombs-ocient) [inline src/main/java/com/ocient/cli/CLI.java:96]: Instead of setting this here, can we just set it in the `main` loop by checking the `lastCommandErrored` bool?

File: src/main/java/com/ocient/cli/CLI.java [hunk 2]
```diff
@@ -1298,6 +1324,9 @@ else if (startsWithIgnoreCase(cmd, "SET TIMEOUT"))
 		{
 			setQueryTimeout(cmd);
 		}
+		else if(startsWithIgnoreCase(cmd, "SET CLI LOGGING")){
```
Comment (dcoombs-ocient) [inline src/main/java/com/ocient/cli/CLI.java:1327]: Lets call this `CLI TRACE` since this isn't really logging (we are just writing to stdout). `CLI TRACE ON` and `CLI TRACE OFF`. It seems like most of these commands (besides timeout) leave out the `SET` portion.

File: src/main/java/com/ocient/cli/CLI.java [hunk 3]
```diff
@@ -937,7 +960,7 @@ public static void main(final String[] args)
 		}
 		catch (final UserInterruptException | EndOfFileException e)
 		{
-			return;
+			System.exit(returnErrorCode ? 1 : 0);
```
Comment (dcoombs-ocient) [inline src/main/java/com/ocient/cli/CLI.java:963]: Can you make the `1` and `0` constants in the code. `EXIT_FAILURE` `EXIT_SUCCESS`, in case we add other exit calls in the future?

**Review by dcoombs-ocient — CHANGES_REQUESTED** [review]

File: src/main/java/com/ocient/cli/CLI.java (hunk 1 above)
Comment (tli-ocient) [reply src/main/java/com/ocient/cli/CLI.java]: thats better. changed

File: src/main/java/com/ocient/cli/CLI.java (hunk 2 above)
Comment (tli-ocient) [reply src/main/java/com/ocient/cli/CLI.java]: ok changed.

File: src/main/java/com/ocient/cli/CLI.java (hunk 3 above)
Comment (tli-ocient) [reply src/main/java/com/ocient/cli/CLI.java]: yes, done.

File: src/main/java/com/ocient/cli/CLI.java [hunk 4]
```diff
@@ -416,7 +421,6 @@ private static void executeAndPrint(final String cmd)
 			{
 			}
 			System.out.println("Error: " + e.getMessage());
-			lastCommandErrored = true;
```
Comment (dcoombs-ocient) [inline src/main/java/com/ocient/cli/CLI.java:419]: Why was this removed?

Comment (tli-ocient) [reply src/main/java/com/ocient/cli/CLI.java]: My bad, accidentally cut when I was editing.

**Review by dcoombs-ocient — APPROVED** [review]
